packages/flutter_tools/lib/src/base/dds.dart

```diff
@@ -59,9 +59,19 @@ class DartDevelopmentService {
     } on dds.DartDevelopmentServiceException catch (e) {
       logger.printTrace('Warning: Failed to start DDS: ${e.message}');
       if (e.errorCode == dds.DartDevelopmentServiceException.existingDdsInstanceError) {
-        _existingDdsUri = Uri.parse(
-          e.message.split(' ').firstWhere((String e) => e.startsWith('http'))
-        );
+        try {
+          _existingDdsUri = Uri.parse(
+            e.message.split(' ').firstWhere((String e) => e.startsWith('http'))
+          );
+        } on StateError {
+          logger.printError(
+            'DDS has failed to start and there is not an existing DDS instance '
+            'available to connect to. Please comment on '
+            'https://github.com/flutter/flutter/issues/72385 with output from '
+            '"flutter doctor -v" and the following error message:\n\n ${e.message}.'
+          );
+          throw e;
```

Comment:
So, we want to both do the `printError`, **and** put more info into the thrown exception so that it will show up in crash logging. I believe `e.message` here will just be "Bad State: no element" which is what we're already getting.

Reply:
`e` here is actually the `DartDevelopmentServiceException`, not the `StateError` (we're not capturing the state error, just catching it).

Reply:
Oh whoops. I misread. I thought `e` was the exception object.

Reply:
It's _an_ exception object, just not the one you thought :-).

Reply:
Yeah, fair point. Done.